packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -805,18 +860,26 @@ jsi::Value JavaTurboModule::invokeJavaMethod(
               throw std::invalid_argument("Promise fn arg count must be 2");
             }
 
+            std::shared_ptr<std::string> jsStack = std::make_shared<std::string>(createJSRuntimeError(runtime, "")
+                .asObject(runtime).getProperty(runtime, "stack").asString(runtime).utf8(runtime));
```

Comment:
+1

I'm not sure why we need to do this, rejecting the promise should be sufficient, and it's up to the JS runtime to provide a good stacktrace for uncaught rejected promises - eg done here: https://github.com/facebook/react-native/blob/f10dd3f6660bbe6347b65e871dd11a2290426abe/packages/react-native/Libraries/Core/polyfillPromise.js#L32

Reply:
The goal is to track down the cause of the rejected promise, which means from where was it called.

Because the promise is rejected asynchronously we won't be able to get the original call stack trace, we only get the current one which will track down to internal timer calls which I think is not useful.

I would like to improve this, as at the moment the JS stack is captured before execution. Therefore for all promise call even if they are resolved. But if we do not save the stack before we will lose the context. 

Example:
![Screenshot 2023-05-23 at 10 08 27](https://github.com/facebook/react-native/assets/31292499/5770f246-a034-42cf-94e3-a48381f02609)

Reply:
Same as on iOS also gated reading of the JS stack here.
https://github.com/krystofwoldrich/react-native/blob/40163ffc18a702519d08df1b3a422392b7d7497b/packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp#L891